iothub/service/src/Amqp/AmqpConnectionHandler.cs

```diff
@@ -250,6 +252,83 @@ private static AmqpSettings CreateAmqpSettings()
             return amqpSettings;
         }
 
+        private async Task<ClientWebSocketTransport> CreateClientWebSocketTransportAsync(CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, cancellationToken, nameof(CreateClientWebSocketTransportAsync));
+
+            try
+            {
+                var websocketUri = new Uri($"{AmqpsConstants.Scheme}{_credential.HostName}:{AmqpsConstants.WebsocketPort}{AmqpsConstants.UriSuffix}");
+
+                if (Logging.IsEnabled)
+                    Logging.Info(this, websocketUri, nameof(CreateClientWebSocketTransportAsync));
+
+                ClientWebSocket websocket = CreateClientWebSocket();
+
+                await websocket.ConnectAsync(websocketUri, cancellationToken).ConfigureAwait(false);
+
+                return new ClientWebSocketTransport(websocket, null, null);
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, cancellationToken, nameof(CreateClientWebSocketTransportAsync));
+            }
+        }
+
+        private ClientWebSocket CreateClientWebSocket()
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, nameof(CreateClientWebSocket));
+
+            try
+            {
+                // If the user provided a client web socket instance, just use it as is.
+                if (_options.ClientWebSocket != null)
+                {
+                    return _options.ClientWebSocket;
+                }
+
+                var websocket = new ClientWebSocket();
+
+                // Set SubProtocol to AMQPWSB10
+                websocket.Options.AddSubProtocol(AmqpsConstants.Amqpwsb10);
+
+                if (_options.WebSocketKeepAlive.HasValue)
+                {
+                    websocket.Options.KeepAliveInterval = _options.WebSocketKeepAlive.Value;
+                }
+
+                // Check if we're configured to use a proxy server
+                IWebProxy webProxy = _options.Proxy;
+
+                try
+                {
+                    if (webProxy != null)
+                    {
+                        // Configure proxy server
+                        websocket.Options.Proxy = webProxy;
+                        if (Logging.IsEnabled)
+                            Logging.Info(this, "Setting ClientWebSocket.Options.Proxy", nameof(CreateClientWebSocket));
+                    }
+                }
+                catch (PlatformNotSupportedException)
+                {
+                    // .NET Core 2.0 doesn't support proxy. Ignore this setting.
+                    if (Logging.IsEnabled)
```

Comment:
I asked this question on Tahir's PR. Should we silently fail if the proxy set fails? Seems to me we should throw an exception.

Reply:
Yeah we should throw here instead

Reply:
Allowing the PlatformNotSupportedException to bubble up to the user seems appropriate here to me. What do you think @tmahmood-microsoft ?

Reply:
Yeah, throwing an exception makes sense. 
I am throwing InvalidOperationException for such case in device side.

Reply:
I'd say throw InvalidOperationException with the PNSE as the inner.